code/game/objects/items/weapons/storage/backpack.dm

```diff
@@ -356,16 +356,71 @@
 	new /obj/item/crowbar(src)
 
 /*
- * Duffelbags - My thanks to MrSnapWalk for the original icon and Neinhaus for the job variants - Dave.
+ * Duffelbags
  */
 
 /obj/item/storage/backpack/duffel
 	name = "duffelbag"
-	desc = "A large grey duffelbag designed to hold more items than a regular bag."
+	desc = "A large grey duffelbag designed to hold more items than a regular bag. It slows you down when you carry it unzipped."
 	icon_state = "duffel"
 	item_state = "duffel"
 	max_combined_w_class = 30
-	slowdown = 1
+	/// Is the bag zipped up?
+	var/zipped = TRUE
+	/// How long it takes to pull things out of this bag
+	var/zip_time = 0.7 SECONDS
+
+/obj/item/storage/backpack/duffel/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>It is currently [zipped ? "zipped" : "unzipped"].</span>"
+	if(zipped)
+		. += "<span class='notice'>Ctrl+click or use the verb to un-zip it!</span>"
+
+/obj/item/storage/backpack/duffel/CtrlClick(mob/user)
+	. = ..()
+	handle_zipping()
+
+/obj/item/storage/backpack/duffel/verb/handle_zipping()
+	set name = "Toggle Duffelbag Zip"
+	set category = "Object"
+	set src in usr
+
+	if(!zip_time || do_after(usr, zip_time, target = src))
+		playsound(src, 'sound/items/zip.ogg', 75, TRUE)
+		zipped = !zipped
+
+		if(!zipped && zip_time) // Handle slowdown and stuff now that we just zipped it
+			slowdown = 1
+			show_to(usr)
+			return
+
+		slowdown = 0
+		hide_from_all()
+		for(var/obj/item/storage/container in src)
+			container.hide_from_all() // Hide everything inside the bag too
```

Comment:
would you just make this a proc and pass the `user` arg to it instead of using `usr`

Reply:
A proc would work, but you would only be able to action by using CTRL+Click, without a right click verb. Is that acceptable?